pkg/start/start.go

```diff
@@ -180,6 +201,7 @@ func (o *Options) Run(ctx context.Context) error {
 			return false, nil
 		default:
 			startingFeatureSet = string(gate.Spec.FeatureSet)
+			enabledGates = cvoGatesFrom(gate)
```

Comment:
now that we're going to start depending on `cluster` FeatureGate `status` here, we'll need something more granular than our current feature-set watcher to notice those changes and keep us up to date.  Maybe `pkg/featurechangestopper` (already watching/polling) should grow an API to call a callback on changes, and we have a lock on `Operator` and  a goroutine that updates `Operator.enabledGates` on changes?  Or rolls the CVO container?  Or...?

Reply:
I mistakenly believed that `featurechangestopper` would kill us on any change but that's really not what happens. library-go's FeatureGateAccess terminates when the set of enabled/disabled gates changes. Our `featurechangestopper` terminates when featureset changes.

I think for our case it could be appropriate to do something like what `FeatureGateAccess` does and simply terminate on observed gate changes, probably check just the ones that CVO cares about.

Reply:
Addressed in c1b27673eaf1d1eb10399589860378565e3324fc by making `featurechangestopper` terminate CVO when CVO-relevant flags change.